tests/test_cli_compile.py

```diff
@@ -2735,9 +2737,372 @@ def test_all_extras_fail_with_extra(fake_dists, runner, make_module, fname, cont
     assert exp in out.stderr
 
 
+def _parse_deps(text: str, keys: Iterable[str]) -> dict[str, dict[str, Any] | None]:
+    """Get structured representation of ``pip-compile`` output.
+
+    :param text: output from ``pip-compile``
+    :param keys: names of expected packages. If not in text the version will be set to ``None``.
+    """
+    result: dict[str, dict[str, Any] | None] = {k: None for k in keys}
+    kwargs: dict[str, Any] = {}
+    for line in text.splitlines():
+        not_comment, _, comment = line.partition("#")
+
+        pkg_name, eq_sep, pkg_version = not_comment.strip().partition("==")
+        if eq_sep:
+            kwargs = {"version": pkg_version}
+            result[pkg_name] = kwargs
+            assert not comment
+            continue
+
+        # Beginning of via section
+        before_via, via_sep, after_via = comment.partition("via")
+        if via_sep:
+            kwargs["via"] = set()
+            if after_via.strip():
+                kwargs["via"].add(after_via.strip())
+            continue
+
+        # Continuation of via section
+        if "via" in kwargs and not_comment == "    ":
+            kwargs["via"].add(comment.strip())
+
+    return result
+
+
+class Dependency:
+    def __init__(self, version: str, via: Iterable[str] | None = None) -> None:
+        self._pat = version
+        self._via = None if via is None else set(via)
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, dict):
+            return False
+
```

Comment:
Since this is not used, maybe we should delete dead code?
```suggestion
```

Reply:
Yeah, I guess a self-test might be useful.

Reply:
Obsolete